Source/WebCore/platform/network/curl/CurlProxySettings.h

```diff
@@ -62,6 +62,10 @@ class CurlProxySettings {
     void setAuthMethod(long);
     long authMethod() const { return m_authMethod; }
 
+    using PlatformDataVariant = std::variant<Mode, std::pair<URL, String>>;
```

Comment:
This is fine, but it makes it possible to send Mode::Custom without a URL/String pair.  A further refinement would be something like this:

struct DefaultData { };
struct NoProxyData { };
struct CustomData {
    URL url;
    String ignoreHosts;
};
using PlatformDataVariant = std::variant<DefaultData, NoProxyData, CustomData>;

Then also use PlatformDataVariant for member storage if you really want to go to town.